jina/executors/encoders/numeric/fit_transform.py

```diff
@@ -0,0 +1,73 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import numpy as np
+
+from .. import BaseNumericEncoder
+from ...decorators import batching
+
+class FitTransformEncoder(BaseNumericEncoder):
+    """
+    :class:`FitTransformEncoder` encodes data from an ndarray in size `B x T` into an ndarray in size `B x D`
+    Parent class for RandomGaussianEncoder, RandomSparseEncoder and TSNEEncoder to prevent redundant code
+    """
+
+    def __init__(self,
+                 output_dim: int,
+                 random_state=2020,
+                 *args,
+                 **kwargs):
+        """
+        :param output_dim: the output size.
+        """
+        super().__init__(*args, **kwargs)
+        self.output_dim = output_dim
+        self.random_state = random_state
+
+    @batching
+    def encode(self, data: 'np.ndarray', *args, **kwargs) -> 'np.ndarray':
+        """
+        :param data: a `B x T` numpy ``ndarray``, `B` is the size of the batch
+        :return: a `B x D` numpy ``ndarray``
+        """
+        return self.model.fit_transform(data)
```

Comment:
I see a problem. Isn't it going to fit the model every time it encodes a `batch`?